index.rst

```diff
@@ -1,61 +1,311 @@
+Image ingestion
+---------------
 
-.. note::
+As part of the wavefront processing commands, the current version of the MTAOS receives files names and directory location.
+This data is initially ingested to a butler instance to allow them to be ISR corrected and later WEP processed.
+
+Since the data will now be ingested by a different component and will be made available to the MTAOS through butler, there is no more need for a local ingestion.
+
+Simplification of wfs process commands
+--------------------------------------
+
+As mention above, the wfs process commands receives file names and path.
+With the data already ingested in the butler the images can be uniquely identified by the ``visitId`` entry in the database.
+That means, all process commands need to be updated to receive the ``visitId`` instead of the files name and path.
+
+At the same time, most of the metadata that is now passed as part of the process commands (``fieldRA``, ``fieldDEC``, ``filter``, ``cameraRotation``) is also available as metadata in the butler and can be removed from the command payload.
+
+Furthermore, through the ``visitId`` one can identify the source of the image (ComCam or LSSTCam), so there is no need for special commands for each one of them.
+
+At the same time, we can differentiate between an Intra/Extra wavefront analysis process and a corner wavefront sensors by the input alone.
+If only one input ``visitId`` then it can be assumed to be a corner wavefront sensor run.
+If, in addition to ``visitId`` an ``extraId`` is given, it can be considered and Intra/Extra wavefront process.
+
+The command would be used as follows:
+
+.. code-block:: py
+
+  from lsst.ts import salobj
+
+  d = salobj.Domain()
+
+  mtaos = salobj.Remote(d, "MTAOS")
+
+  await mtaos.start_task
+
+  # Process full array intra/extra focal images.
+  # Frames can be ComCam or LSSTCam
+  await mtaos.cmd_runWEP.set_start(
+      visitId=2020103000040, extraId=2020103000041
+  )
+
+  # Process corner wavefront sensor
+  # Frames should be LSSTCam only. If ComCam command will raise an exception.
+  await mtaos.cmd_runWEP.set_start(
+      visitId=2020103000040
+  )
+
+Source selection command
+------------------------
+
+In most cases, we will known the coordinates of the next visit field ahead of time.
+That means some pre-processing activities can be done in parallel, while the telescope slew and the image is acquired.
+One of these processes is to prepare a list of suitable targets for wavefront sensing.
+
+This could be done by adding a command that would cause the MTAOS to run the source selection routine.
+The command payload would minimally be;
+
+  - RA: Right ascension of the field (in hours).
+  - Dec: Declination of the field (in degrees).
+  - PA: Position angle (in degrees).
+  - mode: An enumeration specifying the wfs mode; corner, full array, etc.
+
+Issuing the command would look something like:
+
+.. code-block:: py
+
+  from lsst.ts.idl.MTAOS import WFSMode
```

Comment:
Please use the enum of `CamType` in `ts_wep` instead to avoid the duplication.

Reply:
done